src/reconcile/rec_write.c

```diff
@@ -1534,7 +1535,11 @@ __rec_set_page_write_gen(WT_PAGE_HEADER *dsk, WT_BTREE *btree)
      * Other than salvage, the write generation number is used to reset the stale transaction id's
      * present on the page upon server restart.
      */
-    dsk->write_gen = ++btree->write_gen;
```

Comment:
Better to add a comment here to describe this.

Reply:
I dropped the approach of using the same write gen number and update the connection base write generation number.